src/app/product-form/product-form.component.html

```diff
@@ -0,0 +1,64 @@
+<form [formGroup]="productForm">
+  <div class="inputBorder">
+    <mat-form-field>
+      <mat-label>Name</mat-label>
+      <input matInput placeholder="Name of the product" [defaultValue]="productForm.get('name')" formControlName="name">
+    </mat-form-field>
+    <div *ngIf="productForm.controls['name'].errors?.required && !productForm.controls['name'].untouched"><span class="error">Name is required</span></div>
+  </div>
+
+  <div class="inputBorder">
+    <mat-form-field>
+      <mat-label>Description</mat-label>
+      <textarea matInput formControlName="description" [defaultValue]="productForm.get('description')"></textarea>
+    </mat-form-field>
+    <div *ngIf="productForm.controls['description'].errors?.required && !productForm.controls['description'].untouched"><span class="error">Description is required</span></div>
+  </div>
+
+  <div class="inputBorder">
+    <mat-form-field>
+      <mat-label>Price</mat-label>
+      <input matInput placeholder="Price of the product" (keypress)="numberOnly($event)" [defaultValue]="productForm.get('price')" formControlName="price">
```

Comment:
I prefer you used type="number" here rather than that event on keypress. Either that or a mask. This implementation is not that user friendly.